gltf/gltf_system.rs

```diff
@@ -404,9 +406,15 @@ impl GltfModel {
     pub fn draw_scene(&self, scene_id: usize, viewport_dimensions: [u32; 2],
                       mut builder: AutoCommandBufferBuilder) -> AutoCommandBufferBuilder
     {
-        for node in self.gltf.as_json().scenes[scene_id].nodes.iter() {
-            builder = self.draw_node(node.value(), Matrix4::identity(), viewport_dimensions,
-                                     builder);
+        let scene = self.gltf.scenes().nth(scene_id).unwrap();
+        for node in scene.nodes() {
+            let world_to_framebuffer = Matrix4::new(
```

Comment:
The duck was rendered upside down because the model uses the OpenGL co-ordinate system instead of the Vulkan co-ordinate system. The matrix was taken from this source: https://matthewwellings.com/blog/the-new-vulkan-coordinate-system/

Reply:
Yeah this piece of code was a work-in-progress.
There's also the problem of the depth coordinate which is not the same as in OpenGL (you lose half of the precision of you use an OpenGL matrix).